clockwork.go

```diff
@@ -108,6 +125,27 @@ func (fc *fakeClock) After(d time.Duration) <-chan time.Time {
 	return done
 }
 
+// AfterTime, when called with time `t`, is similar to calling `After(t.Sub(Now()))`.
+// Crucially, it is race-free, and therefore can be called from one go-routine
+// while another is calling `Advance`.
+func (fc *fakeClock) AfterTime(t time.Time) <-chan time.Time {
```

Comment:
With the introduction of this method, the previous implementation of `After` could be refactored to just call `fc.AfterTime(now.Add(d))` instead